source/colors.py

```diff
@@ -1,24 +1,26 @@
-#colors.py
-#A part of NonVisual Desktop Access (NVDA)
-#Copyright (C) 2006-2008 NVDA Contributors <http://www.nvda-project.org/>
-#This file is covered by the GNU General Public License.
-#See the file COPYING for more details.
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2006-2019 NV Access Limited, Babbage B.V.
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
 
 from collections import namedtuple
 import math
 import colorsys
 from ctypes.wintypes import COLORREF
 import re
+from typing import Union
 
 class RGB(namedtuple('RGB',('red','green','blue'))):
 	"""Represents a color as an RGB (red green blue) value"""
 
 	@classmethod
-	def fromCOLORREF(cls,c):
+	def fromCOLORREF(cls, c: Union[COLORREF, int]):
 		"""factory method to create an RGB from a COLORREF ctypes instance"""
-		if isinstance(c,COLORREF):
-			c=c.value
-		return cls(c&0xff,(c>>8)&0xff,(c>>16)&0xff)
+		if isinstance(c, COLORREF):
+			c = c.value
+		if (c >> 24) & 0xff:
+			return cls(-1, -1, -1)

```

Comment:
This looks a little suspicious too. It would be good if there was more documentation to explain this. Unit tests would also be nice.